api/controller.py

```diff
@@ -758,29 +756,21 @@ def search(self, lane_identifier):
         query = flask.request.args.get('q')
         library_short_name = flask.request.library.short_name
 
-        language_header = flask.request.headers.get("Accept-Language")
-        if language_header:
-            languages = parse_header(language_header)
-            languages = map(str, languages)
-            languages = map(LanguageCodes.iso_639_2_for_locale, languages)
-            languages = [l for l in languages if l]
-        else:
-            languages = None
-
         facets = load_facets_from_request(
             worklist=lane, base_class=SearchFacets
         )
-        kwargs = dict()
-        if languages:
-            kwargs['language'] = languages
-        kwargs.update(dict(facets.items()))
 
         # Create a function that, when called, generates a URL to the
         # search controller.
+        #
+        # We'll call this one way if there is no query string in the
+        # request arguments, and another way if there is a query
+        # string.
+        make_url_kwargs = dict(facets.items())
```

Comment:
Previously these URLs were not propagating the SearchFacets items. I don't think this made a difference, though, since it would only have affected the `self` link.